From c11a951a24b91f80e109951b0fe2ce418ea70f17 Mon Sep 17 00:00:00 2001
From: Daiki Ueno <dueno@redhat.com>
Date: Tue, 23 May 2017 11:55:25 +0200
Subject: [PATCH] pkcs11: Make CK_RSA_PKCS_OAEP_PARAMS useful

---
 common/pkcs11.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/common/pkcs11.h b/common/pkcs11.h
index 357c9bb..ad1cdec 100644
--- a/common/pkcs11.h
+++ b/common/pkcs11.h
@@ -738,6 +738,12 @@ struct ck_mechanism_info
   ck_flags_t flags;
 };
 
+#define CKG_MGF1_SHA1 0x00000001UL
+#define CKG_MGF1_SHA224 0x00000005UL
+#define CKG_MGF1_SHA256 0x00000002UL
+#define CKG_MGF1_SHA384 0x00000003UL
+#define CKG_MGF1_SHA512 0x00000004UL
+
 typedef unsigned long ck_rsa_pkcs_mgf_type_t;
 typedef unsigned long ck_rsa_pkcs_oaep_source_type_t;
 
@@ -1319,7 +1325,7 @@ typedef struct ck_function_list **CK_FUNCTION_LIST_PTR_PTR;
 typedef struct ck_c_initialize_args CK_C_INITIALIZE_ARGS;
 typedef struct ck_c_initialize_args *CK_C_INITIALIZE_ARGS_PTR;
 
-typedef struct ck_rsa_pkcs_oaep_params CK_RSA_PKCS_OAEP_PARAM;
+typedef struct ck_rsa_pkcs_oaep_params CK_RSA_PKCS_OAEP_PARAMS;
 typedef struct ck_rsa_pkcs_oaep_params *CK_RSA_PKCS_OAEP_PARAMS_PTR;
 
 #define NULL_PTR NULL
-- 
2.9.4

